HelloWorld/HelloWorldImplementation.h

```diff
@@ -0,0 +1,138 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2024 RDK Management
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#pragma once
+
+#include "Module.h"
+#include <interfaces/Ids.h>
+#include <interfaces/IHelloWorld.h>
+#include <websocketpp/config/asio_no_tls_client.hpp>
+#include <websocketpp/client.hpp>
+
+#include <websocketpp/common/thread.hpp>
+#include <websocketpp/common/memory.hpp>
+//#include "tracing/Logging.h"
+
+//#include <com/com.h>
+//#include <core/core.h>
+//#include <plugins/plugins.h>
+
+#define HELLOWORLD_NAMESPACE "HelloWorld"
+
+namespace WPEFramework {
+namespace Plugin {
+
+
+    class HelloWorldImplementation : public Exchange::IHelloWorld {
+
+    public:
+        HelloWorldImplementation();
+        ~HelloWorldImplementation() override;
+
+        // We do not allow this plugin to be copied !!
+        HelloWorldImplementation(const HelloWorldImplementation&) = delete;
+        HelloWorldImplementation& operator=(const HelloWorldImplementation&) = delete;
+
+        BEGIN_INTERFACE_MAP(HelloWorldImplementation)
+        INTERFACE_ENTRY(Exchange::IHelloWorld)
+        END_INTERFACE_MAP
+
+    public:
+
+        enum EventNames {
+          ECHOMESSAGE
+        };
+        class EXTERNAL Job : public Core::IDispatch {
+        protected:
+             Job(HelloWorldImplementation *helloWorldImplementation, EventNames event, JsonValue &params)
+                : _helloWorldImplementation(helloWorldImplementation)
+                , _event(event)
+                , _params(params) {
+                if (_helloWorldImplementation != nullptr) {
+                    _helloWorldImplementation->AddRef();
+                }
+            }
+
+       public:
+            Job() = delete;
+            Job(const Job&) = delete;
+            Job& operator=(const Job&) = delete;
+            ~Job() {
+                if (_helloWorldImplementation != nullptr) {
+                    _helloWorldImplementation->Release();
+                }
+            }
+
+            typedef websocketpp::client<websocketpp::config::asio_client> client;
```

Comment:
This is matching code from a websocket tutorial and needs to be credited.  1. Please insert here:
```
Copyright (c) 2014, Peter Thorson. All rights reserved.
Licensed under the BSD-3 License
```

2. Please append the same 2 lines to top level NOTICE
3. Please append a copy of the BSD-3 License to top level LICENSE.  (Unfortuately there isn't one already.)
4. This may require recipe checksum updates.